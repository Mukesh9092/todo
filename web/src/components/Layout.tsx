import * as React from "react";

import {
  AppBar,
  Container,
  Toolbar,
  Typography,
  CssBaseline,
} from "@material-ui/core";

export interface LayoutProps {
  children: React.ReactNode;
}

const Layout: React.SFC<LayoutProps> = (props) => {
  const { children } = props;

  return (
    <>
      <CssBaseline />
      <AppBar color="primary" position="static" style={{ height: 64 }}>
        <Toolbar style={{ height: 64 }}>
          <Typography color="inherit">Todo App</Typography>
        </Toolbar>
      </AppBar>
      <Container maxWidth="sm">{children}</Container>
    </>
  );
};

export default Layout;
