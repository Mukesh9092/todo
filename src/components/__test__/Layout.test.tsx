import { shallow } from 'enzyme';
import * as React from 'react';

import Layout from '../Layout';

describe('[Component] Layout', () => {
  it('renders without error', () => {
    const wrapper = shallow(<Layout>Boom</Layout>);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
