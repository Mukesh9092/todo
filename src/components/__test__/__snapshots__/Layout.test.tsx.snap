// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Layout renders without error 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2",
      "style": "padding:0;margin:0;background-color:#fafafa",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "MuiPaper-root-1 MuiPaper-elevation4-7 MuiAppBar-root-28 MuiAppBar-positionStatic-32 MuiAppBar-colorPrimary-35",
          "style": "height:64px",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "MuiToolbar-root-37 MuiToolbar-regular-39 MuiToolbar-gutters-38",
              "style": "height:64px",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "MuiTypography-root-41 MuiTypography-body1-50 MuiTypography-colorInherit-70",
                },
                "children": Array [
                  Object {
                    "data": "TODO APP",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "style": undefined,
            },
          },
        ],
        "name": "header",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "data": "Boom",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "text",
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
      Object {
        "data": "Boom",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "MuiPaper-root-1 MuiPaper-elevation4-7 MuiAppBar-root-28 MuiAppBar-positionStatic-32 MuiAppBar-colorPrimary-35",
            "style": "height:64px",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "MuiToolbar-root-37 MuiToolbar-regular-39 MuiToolbar-gutters-38",
                "style": "height:64px",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "MuiTypography-root-41 MuiTypography-body1-50 MuiTypography-colorInherit-70",
                  },
                  "children": Array [
                    Object {
                      "data": "TODO APP",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "p",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "style": undefined,
              },
            },
          ],
          "name": "header",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
